M2/Macaulay2/m2/testing.m2

```diff
@@ -73,9 +63,14 @@ loadTestDir := pkg -> (
     if fileExists testDir then (
         tmp := currentPackage;
         currentPackage = pkg;
-        TEST(sort apply(select(readDirectory testDir, file ->
-            match("\\.m2$", file)), test -> testDir | test),
-            FileName => true);
+	scan(sort select(readDirectory testDir, file -> match("\\.m2$", file)),
+	    test -> (
+		testpath := testDir | test;
+		teststr := get testpath;
+		testlines := lines teststr;
+		addTest(teststr,
+		    new FilePosition from(testpath, 1, 1,
+			#testlines, #testlines#-1))));
```

Comment:
This seems range here. Maybe add another method to `addTest` that takes a filename and call it here? And in that method, I don't think you need to store an actual `FilePosition`, just the string filename (possibly with ":0:0" appended) should suffice. This is actually a good reason why having `FilePosition` with only a start point is useful.